test/http/request.cpp

```diff
@@ -3,83 +3,204 @@
 // information on the copying conditions.
 
 #define CATCH_CONFIG_MAIN
-#include "src/ext/Catch/single_include/catch.hpp"
 
+#include "src/common/check_connectivity.hpp"
+#include "src/ext/Catch/single_include/catch.hpp"
+#include "src/http/request_impl.hpp"
 #include <measurement_kit/common.hpp>
+#include <measurement_kit/ext.hpp>
 #include <measurement_kit/http.hpp>
-
-#include "src/common/check_connectivity.hpp"
-#include "src/http/request.hpp"
+#include <openssl/md5.h>
 
 using namespace mk;
 using namespace mk::net;
 using namespace mk::http;
 
-// TODO: after refactoring there are probably duplicated tests, we
-// should merge them to avoid testing things twice
+// Either tor was running and hence everything should be OK, or tor was
+// not running and hence connect() to socks port must have failed.
+static inline bool check_error_after_tor(Error e) {
+    return e == NoError() or e == ConnectFailedError();
+}
+
+static std::string md5(std::string s) {
+    static const char *table[] = {"0", "1", "2", "3", "4", "5", "6", "7",
+                                  "8", "9", "a", "b", "c", "d", "e", "f"};
+
+    unsigned char digest[MD5_DIGEST_LENGTH];
+    MD5((const unsigned char *)s.data(), s.size(), digest);
+    std::string retval;
+    for (int i = 0; i < MD5_DIGEST_LENGTH; ++i) {
+        retval += table[((digest[i] & 0xf0) >> 4)];
+        retval += table[(digest[i] & 0x0f)];
+    }
+    return retval;
+}
```

Comment:
Added because I want to make more assertive tests. Previously, as noted by @AntonioLangiu, too many tests in the `http` suite were not performing assertions, which is bad.